### ISSUE TITLE
Gift Cards with Decimals and localization using comma separator instead of period

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#1694]
### Issue / Question / Bug

Before submitting an issue please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] I'm reporting an issue of an unmodified OSPOS installation
- [x] I checked open and closed issues database and no similar issue was already discussed (please make sure you searched!)
- [x] I read the README, WHATS_NEW and UPGRADE
- [x] I read the FAQ (https://github.com/jekkos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP has php-gd, php-intl, sockets and etc. installed)
- [x] I read the wiki
- [x] I ran any database upgrade scripts (e.g. database/2.4_to_3.0.sql), and migrating function
- [x] I'm aware the latest master could be a development version and therefore not stable
- [x] I know the version of OSPOS and git commit hash (check the footer of your OSPOS), the name and version of OS, Web server, PHP and MySQL and will add them to my issue report

### Installation information

- OSPOS version is: 3.2.0
- OSPOS git commit hash is: 4f5ad57
- PHP version is:   (e.g. 5.5, 5.6, 7.0, 7.1) 5.6.32
- MySQL or MariaDB version is:   (e.g. MySQL 5.5, MySQL 5.6, MySQL 5.7, MariaDB 10.0, MariaDB 10.1, MariaDB 10.2) MySQL 5.7
- OS and version is:   (e.g. CentOS 6.9, Ubuntu 16.4, Windows 10) FreeBSD 11.0
- WebServer is:   (e.g. Apache 2.2, Apache 2.4, Nginx 1.12, Nginx 1.13) Apache 2.4
- (If applicable) Installation package for the LAMP/LEMP stack is:   (e.g. WAMP, XAMPP)

### Expected behaviour
When using a localization such as az-AZ or ru that uses a x.xxx,yy number format instead of x,xxx.yy changing or adding a gift card involving decimals would result in an updated amount or an error message indicating the format is incorrect or would properly update the database.

### Actual behaviour
The data is saved in the database as 0.

### Steps to reproduce the issue
If x.xxx,yy is entered for the gift card amount when using a locale that uses that format an error appears "Gift Card Value must be a number." 
![error with correct formatting](https://user-images.githubusercontent.com/17935339/31931453-5a77206c-b8b4-11e7-93e3-ba7cd7777148.JPG)

If x,xxx.yy format is used it sends the message "You have successfully updated Gift Card ...", however it then shows a zero balance of the gift card rather than the number previously entered.
![result of incorrect format](https://user-images.githubusercontent.com/17935339/31931533-8dd5fe1a-b8b4-11e7-9a56-22785a520d85.JPG)

There are no error logs generated.  I turned on database logs and here is the difference in the two update scripts.

When entering 231 with no period or comma
`UPDATE 'ospos_giftcards' SET 'record_time' = '2017-10-24 11:53:56', 'giftcard_number' = '0514230362', 'value' = 231, 'person_id' = '26'`
`WHERE 'giftcard_id' = '654354382'`
`Execution Time:0.0484459400177`

So, you can see there is no problem and the database gets updated properly.

When entering 231.30 (231,30 produces an error)
`UPDATE 'ospos_giftcards' SET 'record_time' = '2017-10-24 11:54:05', 'giftcard_number' = '0514230362', 'value' = 0, 'person_id' = '26'`
`WHERE 'giftcard_id' = '654354382'`
`Execution Time:0.029716014862061`

You can see the database query is updating it to 0 which means this is a code problem rather than a database problem.  There are two possible appropriate behaviors here.  The strict behavior should be to require the exact same locale format in the gift card entry field that the locale is set to, then update the database accordingly or error when the format is incorrect.  The permissive behavior would be to allow formatting in x,xxx.yy or x.xxx,yy format and update the database accordingly or error if not one of those formats.

The quickest fix on this would be to update our error checking code to allow the locale format and not expect x,xxx.yy every time.

**Comment by objecttothis** [comment]
I have confirmed that if I change the locale to us_US for example, and then change the gift card amount it works without problems.  For now I will make this a workaround.  Since I am about to go on sabbatical, I won't be able to finish this code change before the 3.2.0 release deadline of October 31st.  Either someone else could make the code change or they can wait until I get on location and I will be able to change it by the 3.3.0 release deadline, but first if someone else could confirm they can replicate the issue that would be helpful.

**Comment by jekkos** [comment]
Hm the release deadline might just be an indicative one. I think we just want to release once we have something done and stable. But did you say then that the format does work elsewhere but not in giftcards?
Or is this problem exclusively in giftcards only? Might be another wrong locale for AZ_az then in the end? Or am I missing something

**Comment by daN4cat** [comment]
Release date in indicative indeed.

**Comment by objecttothis** [comment]
I haven't tested outside of giftcards.  I will test others when I get a chance.  This should be something that anyone can recreate using a locale that uses x.xxx,yy format.  I think fr_FR does, but can't remember.  ru_RU does for sure.

**Comment by objecttothis** [comment]
OK, I'm working on this and should have something ready/tested before the 1st.

**Comment by daN4cat** [comment]
Oh great to see you back here.

**Comment by daN4cat** [comment]
When you have time countercheck Russian translations in Weblate.

**Comment by objecttothis** [comment]
@daN4cat I checked Russian as you requested and created a ticket related to it and Azerbaijani.  Essentially we need to pull weblate into the master for those two languages as the master is stale.  Russian and Azerbaijani still have completions needs and I have native speakers making progress but it would be nice to have the most up to date data for 3.2.0.

@jekkos @daN4cat DECISIONS

- we need to decide how we want to handle the data for the issue above.  Do we want the database to always store x,xxx.yy format regardless of whether the localization is x.xxx,yy or not?  If we store it depending on the localization that can create inconsistencies in the data if the store localization is changed at a later time or (more likely) data migrated where the user begins using a new localization.  If we always store it in x,xxx.yy (actually I think the default is without the thousands separator) then we just need to "clean" the data before making database updates so that if the decimal separator is a comma, we catch and convert it to a period.

- Second we need to decide whether we use a permissive or strict ECC in data input.  As described above, strict ECC will require the user to use the number format for the locale that they have in the settings or give an error.  Permissive ECC would error if entry is NaN but would allow either x,xxx.yy or x.xxx,yy formatting to proceed.

- Finally, we need to decide if we want to finish this before releasing 3.2.0 or punt to 3.3.0.  I am OK with either.  My wife and kids just got back and they have been jet-lagging which has slowed my work on this, but it's probably reasonable to finish in a few days once the above have been decided.  Since this issue has a reasonable workaround, I don't feel strongly about it making it into 3.2.0.

Ok, the ball is in your court.  Thoughts?

**Comment by daN4cat** [comment]
Weblate: @jekkos please can you look into this?
We need to rethink this translation story as it stands it doesn't work as expected.

Data format: the database representation should be independent from the localisation one. So xxxx.xx.

Sanity check at input time (front end), pull the localisation schema from backend.

I'll postpone 3.2.0 release because there is a blocking issue that I need to sort out and I need to fix another problem.
So we have another month to include translations, fixes and your change here.

Thanks for the help.

**Comment by jekkos** [comment]
Regarding weblate, think the main problems were before I patched it with the translate toolkit. It does involve a bit of manual setup for a new language.. but after that translating seems to be working well.

It's just that I need to find a way to sync some of the older work with the git repo. Will have give it another try today.

Agree on data format. Should be locale agnostic. There's already quite some translations added so let's keep this going.

**Comment by objecttothis** [comment]
OK, I will submit a pull request then to have the code clean the formatting before submitting a query.  As for user entry, do I understand properly that the sanity check will require the same formatting that the locale uses (strict) rather than either or (permissive)?

**Comment by daN4cat** [comment]
I would say, yes sanity check at entry level.

**Comment by daN4cat** [comment]
@objecttothis any progress?

**Comment by objecttothis** [comment]
So it appears the problem I'm having is that jQuery validate does not, by default, see nnn,dd as a number, so I'm trying to change it to look for a pattern instead, but for some reason its passing the regex pattern through rather than the actual value entered into the form. I'll be on it again Monday.

**Comment by objecttothis** [comment]
OK, I'm stuck.  jQuery Validate seems to be causing problems when replacing "number:" with "pattern:" I've updated /views/giftcards/form.php line 120 to read `pattern: /^-?(?:\d+|\d{1,3}(?:[\s\.,]\d{3})+)(?:[\.,]\d+)?$/` which I've tested at https://regexr.com/ and the regex recognizes both nnn,dd and nnn.dd.  There is nothing generated in the php or query logs, but the console in Firefox Developer Edition reports the following:

![capture](https://user-images.githubusercontent.com/17935339/36366455-a19b6a92-1567-11e8-9318-baa5293ac5a9.JPG)

@jekkos @daN4cat are we using the most recent jQuery Validate plugin?  I looked at the code in /public/dist/opensourcepos.min.js and I see the reference to the number method but not pattern. Perhaps we have an old or non-standard jQuery validate plugin? https://jqueryvalidation.org/

**Comment by daN4cat** [comment]
@objecttothis current version is "jquery-validate": "~1.13.1" (see bower.json).

Which version would you need?
I can check the latest version working with JQuery 2.1.4.

**Comment by daN4cat** [comment]
OK, I see there is a 1.17.0 version: https://github.com/jquery-validation/jquery-validation

Let me check where validation is used and give a try in updating it.

**Comment by daN4cat** [comment]
It's used across all the forms to validate the input.... so it should be easy to see if we have some "panic time" :-)

**Comment by daN4cat** [comment]
@objecttothis BTW, I have in mind to add this: https://openexchangerates.github.io/accounting.js/ because I would need it for another feature. For currency / numeric things it should come handy.

**Comment by daN4cat** [comment]
Here we go in master now.

@objecttothis wait 15 min before you can find the latest master build in bintray.
I added even accounting.js in case we need to make any use in the future.

**Comment by jekkos** [comment]
@objecttothis 
* It might be easier to validate formatted decimals in the backend using an ajax call. This will save you the pain of redoing the configured parsing logic in javacsript as its already available in php itself.
* Another argument to do it this way is security: beside having an obiquitous frontend validation I think it's always a good idea to validate in the backend as well
* Also this approach will be more consistent with other parts of the application

There's plenty of examples that do validation using jquery-validate using an ajax call throughout the code.. if you need some help or specific example, let me know

**Comment by daN4cat** [comment]
Yep agree with @jekkos , I should have thought a bit more about this feature here. Let's keep it consistent with the rest of the code and leverage backend validation so locale support is just in one place.

**Comment by objecttothis** [comment]
@jekkos What you are saying makes sense, however due to the fact that jquery-validate is doing validation at the form level and that doesn't accept a nnn,dd format, we will need to modify it to either do no validation at the front end and all the validation at the back end or do validation at the back end and the front end.  I will test the latest master to see if that resolves the problem with number.  if it does then I will leave the form code the way it is and just do the validation at the back end.

**Comment by daN4cat** [comment]
If I'm not mistaken you can use the js validation to call the backend in an ajax call and check the input.

**Comment by jekkos** [comment]
@daN4cat yes indeed that's the way I'd suggest to do this

**Comment by objecttothis** [comment]
OK, super I will do that.  If someone happens to know a place in the code where js validation is being used to do this with an ajax call as an example just post the file name and line numbers.  Otherwise I will find it myself.  Shouldn't take me too long to replicate.

**Comment by daN4cat** [comment]
I'll look up an example later.

**Comment by daN4cat** [comment]
Check views/configs/locale_config.php:

> 	$('#locale_config_form').validate($.extend(form_support.handler, {
> 		rules:
> 		{
> 			number_locale:
> 			{
> 				required: true,
> 				remote: $.extend({
> 						url: "<?php echo site_url($controller_name . '/check_number_locale')?>",
> 						type: "POST"
> 					}, {
> 					data: $.extend(csrf_form_base(), {
> 						"number_locale" : function() {
> 							return $("#number_locale").val();
> 						},
> 						"thousands_separator": function() {
> 							return $("#thousands_separator").is(":checked");
> 						}
> 					}),
> 					dataFilter: function(data, dataType) {
> 						setup_csrf_token();
> 						var response = JSON.parse(data);
> 						$("#number_locale_example").text(response.number_locale_example);
> 						$("#currency_symbol").val(response.currency_symbol);
> 						$("#thousands_separator").prop('checked', response.thousands_separator);
> 						return response.success;
> 					}
> 				})
> 			}
> 		},

**Comment by mckaygerhard** [comment]
lasted commit ee55be347d69ae6be09174b45ab25cdf6e0f8346 broke the localization possibility of change the languaje of the system, and now its broklen and ajust to only server to english, pof course, if avoid that report made by me, later all of you will be fuerts issues related from others not me !

**Comment by daN4cat** [comment]
@objecttothis Jquery validator is now version 1.14.0 as post 1.15.0 break the code I hinted earlier.

By the way I simplified the code to this in the latest version:

```
	$('#locale_config_form').validate($.extend(form_support.handler, {

		rules:
		{
			number_locale:
			{
				required: true,
				remote: {
					url: "<?php echo site_url($controller_name . '/ajax_check_number_locale')?>",
					type: 'post',
					data: $.extend(csrf_form_base(), {
						'number_locale': $("#number_locale").val(),
						'thousands_separator': $("#thousands_separator").is(":checked")
					}),
					dataFilter: function(data) {
						setup_csrf_token();
						var response = JSON.parse(data);
						$("#number_locale_example").text(response.number_locale_example);
						$("#currency_symbol").val(response.currency_symbol);
						$("#thousands_separator").prop('checked', response.thousands_separator);
						return response.success;
					}
				}
			}
		},

		messages:
		{
			number_locale: {
				required: '<?php echo $this->lang->line('config_number_locale_required') ?>',
				number_locale: '<?php echo $this->lang->line('config_number_locale_invalid') ?>'
			}
		},

		errorLabelContainer: "#locale_error_message_box"
	}));
```

**Comment by objecttothis** [comment]
Understood. I'll submit my pull request Tuesday after testing my code. We have a store-wide inventory tomorrow so probably can't look at it until tomorrow evening.

**Comment by objecttothis** [comment]
Unless I'm missing something the database currently does not store the decimal separator.  I think it just assumes that the decimal separator will be a period.  How do we want to proceed with this?  We could pull the decimal separator symbol in when the locale is saved and store it in app_config along with the rest of the locale information then pull it when doing our validation in the ajax call.  The alternative would be to look up the ICU decimal separator at validation time, but I think that will give much more overhead than just looking up in the database.  Further, adding it in the database allows us to make it customizable (a la currency symbol) by the administrator in case they don't want the decimal separator that the locale auto-populates for them.

**Comment by objecttothis** [comment]
Sorry guys, I was expecting this to be easier to resolve than it has been. It doesn't help that I'm burning the candle at both ends with writing code and trying to resolve store management issues.  I've got code working to accept nnn,dd format when the locale format is set to such and nnn.dd when it isn't however I am hardcoding the decimal separator until we figure out the best way to either store the decimal separator in app_config or pull it from ICU. Also using the example code from locale_config.php it does not produce an error when using the wrong format.  It just updates the giftcard amount to 0.00 or 0,00 depending on the system number format.  I think I just need to tweak my code slightly to get that effect.  I'll be back at it in the morning (Feb 28th) and should have the day to hash out the rest of the code, but let me know @jekkos and @daN4cat about your thoughts on whether we store the decimal separator or just pull it from ICU every time?  I'd prefer to store it in the database but didn't want to make dB changes without talking with you guys first.

**Comment by SteveIreland** [comment]
That's what I say about all of my changes ("it sure looked easier when I committed to do it").

**Comment by objecttothis** [comment]
Haha.  Yeah.  Once I hear back from @daN4cat and @jekkos I should be able to finish up, but I'm trying to work out the kinks now.

**Comment by objecttothis** [comment]
@daN4cat in short PHP Intl gets it's information on locales, languages, formats and currencies from ICU which in turn takes most of its information from CLDR.

http://php.net/manual/en/class.locale.php

**Comment by objecttothis** [comment]
Think the decision that needs to be made for this issue to move forward is whether we should store the decimal symbol for the locale in the database like we do other locale information or if we should pull it each time.  I think it's fewer resources if we store it in the database.

**Comment by daN4cat** [comment]
We should rely on the locale support and not store all these details in the database.

**Comment by daN4cat** [comment]
Outside the currency symbol the thousand separator is only if the user wants it or not without specifying dot or comma. So the fractional should follow the same, it's not stored in the db.

**Comment by objecttothis** [comment]
So I take it you don't think the decimal separator should be customizable?

**Comment by daN4cat** [comment]
No because we don't customise the thousand separator neither.

**Comment by objecttothis** [comment]
Here is where it currently stands.  The code is mostly working but there is something wrong with it.  I cannot get it to behave properly with checking and error reporting when the format is nnn,dd and the locale requires nnn.dd or vis a versa.  with the correct formatting it accepts the change but then with incorrect formatting it accepts it and truncates the change to remove the decimals in the database update.  Alternately I've also experienced it where it says nothing is a number, when it clearly is formatted correctly.  Perhaps another set of eyes will spot something I missed in my code.

**Comment by daN4cat** [comment]
Code is in master now. I just squashed it and made two small amendments.

**Comment by daN4cat** [comment]
@objecttothis I made a couple of tests but just to see is it works fine. Please check with your locale and confirm before we can close this.

**Comment by objecttothis** [comment]
So, bintray doesn't have a build for the last 8 days.  Is something broken in the build process?

**Comment by objecttothis** [comment]
I copied the changes as best I could... I don't think I missed anything.  However the behavior is the same as before.  If the locale requires nnn.dd format and you enter nnn,dd format then it will make the change but in the database it truncates after the decimal place.  For example if I am in the en_US locale and I edit a giftcard with the value set to 15,44 (comma) it will submit successfully but says that the new value of the giftcard is 15.00.  If I enter 15.00 (period) it submits and updates properly to 15.44.  The same is true for az_AZ only the opposite (comma works, but decimal truncates).  I think that this is the nature of parse_decimal in that it parses either and spits out the

**Comment by objecttothis** [comment]
OK, I just submitted a change to correct this problem.  I think we probably want to integrate it into parse_decimal() since we will always want to allow only the correct decimal operator.  Let me know if my hunch is correct and I'll make the change.

**Comment by daN4cat** [comment]
Do you get back at UI level the correct value 15,44 or you get back 15?

**Comment by daN4cat** [comment]
All the latest changes are in master now. @objecttothis please test and confirm it's fine.

**Comment by daN4cat** [comment]
@objecttothis the fix you added doesn't work or I added something silly. Unfortunately I don't have time to fix it now, so please look into this tomorrow.

**Comment by objecttothis** [comment]
OK, let me look into it.

**Comment by daN4cat** [comment]
I simply input £50.00 and it flags as not valid (en_GB locale).

**Comment by objecttothis** [comment]
ok, i'm installing the latest master on my server now and I'll see if I can replicate it.  What browser are you using?

**Comment by daN4cat** [comment]
Chrome

**Comment by objecttothis** [comment]
@daN4cat I just make a commit to the branch (it's not in the master) that moves the check for the proper decimal separator into parse_decimals (locale_helper.php) and I tested it in Firefox Developer's Edition, Internet Explorer and Edge browsers.  All behaved properly.  @daN4cat can you tell me if you are still having issues?  I couldn't replicate the issue, but it might be specific to Chrome since the view depends on client-side input.  In that case we will need to isolate exactly what the issue is.  I did set my locale to en_GB, but could not get the same flag.

**Comment by daN4cat** [comment]
@objecttothis no it doesn't work.
I show you printing the values before the if statement you moved to locale_helper what happens:

[13-Mar-2018 21:26:33 Europe/London] $number: 50.00
[13-Mar-2018 21:26:33 Europe/London] $fmt->parse($number): 50
[13-Mar-2018 21:26:33 Europe/London] strval($fmt->parse($number)): 50
[13-Mar-2018 21:26:33 Europe/London] strval($number): 50.00

**Comment by daN4cat** [comment]
It could be that the issue is PHP version based. Please check that.

**Comment by objecttothis** [comment]
OK I'll look into it. From your output I can see that parse is truncating the decimal. I wonder if you get the same behavior with 50.01. If so then this is an issue of setting the precision. If not then it seems to be something with parse truncating trailing zeros. Can you try 50.01 and tell me what you get?

**Comment by daN4cat** [comment]
I cannot test until this evening, but this is my configuration in case that could be meanwhile of help:

![image](https://user-images.githubusercontent.com/636048/37394354-746fac3e-276b-11e8-9701-26d51defc496.png)

**Comment by objecttothis** [comment]
OK, I was able to replicate it right away.  
50 success
50.00 fail
50.01 success
50,01 fail 

So as you can see, there is some sort of bug with the way that parse is handling 50.00 so that it truncates 50.00 to 50.  So this is not PHP version dependent (since I am using 5.6.32 and you are using 7.1) and it isn't browser dependent (since I am using FF and you Chrome).  I'll figure it out.

**Comment by objecttothis** [comment]
OK, I posted the question on Stack Overflow (https://stackoverflow.com/questions/49277321/php-numberformatterparse-truncates-decimals) so we will see what comes of it.

**Comment by daN4cat** [comment]
@objecttothis you got an answer already. I don't think the comparison is doing what you expect but I didn't have the time to think about this more than the 5 min of test last eve.

You need to have use cases good and bad and figure out the right solution... a bit generic and wholly but that's what it should be always.

**Comment by objecttothis** [comment]
I just committed a change that resolves the problem.  Please confirm working on your end.  The only question for me is if I should be getting the number of decimal places from currency_decimals or cash_decimals.  I don't know the difference.

**Comment by objecttothis** [comment]
gah!  ok, with the latest commit, it fixes the issue of 50.00 being a problem but causes a problem when no decimal and trailing fraction is entered (e. g., 50).  I'm so annoyed at this and all the acrobatics that numberformatter:parse is requiring.

**Comment by daN4cat** [comment]
Well.... have fun

**Comment by objecttothis** [comment]
Based on the responses on stackoverflow it seems like abandoning the use of NumberFormatter::parse() to check for the proper use of decimal symbols and thousands separator and instead shift to treating them like text string and validating that aspect as a text string.  parse would still be used like it was before, but I would likely need to switch to regex for the decimal separator/thousands separator.  Thoughts?

**Comment by jekkos** [comment]
What you state in the stack overflow question, maybe it does parse the comma instead of decimal because one is a thousands or decimal separator or vice versa?

About the truncation. Can't you just run the parsed result through the number formatting again so that the correct amount of decimals are added (as configured in the store?) In that case the final oiutput is the same as the validated input (in terms of formatting)

**Comment by jekkos** [comment]
So first you do the parsing, then reformat using to_decimals which will then add the correct amount of zeroes again.. that simple

**Comment by objecttothis** [comment]
I'm not sure that will make the difference.  Let me explain.  With the regex below when I have the thousands separator unchecked all of my use cases pass, but with thousands separator checked entering 1,234.00 passes but the value in the database gets entered as 1.00, so I don't see how using to_decimal is going to fix that.  Using parse isn't too much different unless I'm just taking crazy pills (which might be the case also).  I'm afraid I don't have a good way to step through the PHP and see the value of my variables as I step through the code, so if you have any suggestions on how you see variables in PHP to debug code let me know.

[code]
//Account for thousands separator
	if(empty($config->item('thousands_separator')))
	{
		$fmt->setAttribute(\NumberFormatter::GROUPING_SEPARATOR_SYMBOL, '');
	}
	$fmt->setAttribute(\NumberFormatter::DECIMAL_SEPARATOR_SYMBOL, ',');

//Prepare regex pattern	
	switch($fmt->getAttribute(\NumberFormatter::DECIMAL_SEPARATOR_SYMBOL))
	{
	    case '.':
	        $decimal_separator = '\.';
            break;
	    case ',':
	        $decimal_separator = ',';
            break;
	}

	switch($fmt->getAttribute(\NumberFormatter::GROUPING_SEPARATOR_SYMBOL))
	{
		case '.':
			$grouping_separator = '\.';
			break;
		case ',':
			$grouping_separator = ',';
			break;
		case ' ':
			$grouping_separator = ' ';
			break;
	}

    $pattern = "/^-?(?:\d+|\d{1,3}(?:.".$grouping_separator."\d{3})+)(?:".$decimal_separator."\d{0,2})?$/";
		
	try
	{
	    if(preg_match($pattern,$number) != 1)
	    {
	        $number = FALSE;
	    }
	    //return $fmt->parse($number);
	    return $number;
	}
	catch (Exception $e)
	{
		return FALSE;
	}
[/code]

use cases for en_US
50		PASS
50.01		PASS
50.00		PASS
1,23		FAIL
1,00		FAIL
abc		FAIL
[blank]		FAIL
1,234.00	PASS (IF THOUSANDS SEPARATOR) 
1,234.56	PASS (IF THOUSANDS SEPARATOR)
1.234,00	FAIL 
1.234,56	FAIL
1.234.56	FAIL

**Comment by daN4cat** [comment]
A simple way of debugging is to read the error_log file after you dumped something.
Example to add to your code: error_log('$number: ' . $number);

**Comment by daN4cat** [comment]
I reverted the fix that was breaking the Giftcard ajax call and in any case many other functions relying on the helper function.

**Comment by objecttothis** [comment]
For now, my recommendation is to release 3.2.0 without this fix as there is a work around.  We just started the process of moving our bookstore location and I don't want to hold up the release any longer.  I will continue to work on a fix and submit something once I have a fix tested with a wide range of use cases and it doesn't break other stuff.

**Comment by daN4cat** [comment]
Makes sense, I was wondering about the same. We move this to a next release and when a sounding solution is found.

**Comment by jekkos** [comment]
Hm would it be that hard to fix this one? I'm willing to give this another look this weekend otherwise.

**Comment by objecttothis** [comment]
@jekkos if you want to give it a shot, you are more than welcome.  It may be something simple, and as far as I'm concerned I'm glad to learn from you guys.  Otherwise I will tackle it and get it working.

**Comment by jekkos** [comment]
Guess this one was fixed by applying the backend locale parsing.. so will close
